### DISCUSSION TITLE
Allow GitHub to use code snippets stayed on when I remember very well turning it off.

### DISCUSSION THREAD

**Discussion by christopherbii** [community/community#66160] [Copilot Conversations]
### Select Topic Area

General

### Body

As the title states. I remember clearly when it asked me if I wanted to let GitHub use my code for improvements and I most definitly chose not to. I remember this was the last submittable step before it was ready to go. Yet today I decided to turn copilot back on for some repetitive tasks and to my suprise, code snippet sharing was turned on.

Anyone else have the same thing happen to them?

If you want to check: Settings > Copilot

This is most likely something wrong that happened on my end, although I am certain I chose to have it off. But am curious to see if it is truly just me and maybe a few others or this true for a lot of you.

**Comment by Akash1134** [comment]
Hi @BiiChris ,

Thanks for posting in the GitHub Copilot community and sorry for the problem caused. Do you remember when did you setup Copilot and disabled _prompt and suggestion collection_? Did you updated/reinstalled Copilot or your IDE in the meanwhile?

We haven't seen any reports about this before so this issue is indeed pretty new for us. Ideally, once option to [collect prompt & suggestion ](https://docs.github.com/en/copilot/configuring-github-copilot/configuring-github-copilot-settings-on-githubcom#enabling-or-disabling-prompt-and-suggestion-collection)is disabled, it should stay that way.

**Reply by christopherbii** [reply]
Hi, sorry for the late response. I setup copilot on the 29th of august. I do not remember uninstalling it or updating it after initial install. Yeah I believe it does and the only reason I brought this up was because I clearly remembered seeing it and disabling it when setting it up. If it cannot be replicated and nobody else has had this issue then it must have been something on my end I guess.

**Reply by dvadneau** [reply]
I have the same issue.

I installed copilot extension for the first time in a new codespace, created for a pair programming with AI course.

I then signed up for the free trial and also specifically remember unchecking the _prompt and suggestion collection_ checkbox.

I had a re-think about the _Suggestions matching public code_, so I visited the https://github.com/settings/copilot page and the checkbox mentioned above is checked.

The thing is, the form won't let me make any changes. Making changes, or even ONLY clicking Save always results in the form error "Select an option below to continue."

I've tried refreshing, logging out and back in, even canceling the trial (thinking I could sign-up again as a fresh start, but it just disables the charge at the end), but the form error is always the answer to clicking Save.

Should I create a new discussion topic for this?

**Reply by Tenrys** [reply]
I'm having the same issue as above.

**Reply by dvadneau** [reply]
I just tried this morning and the form appears to be working. No errors and the values are maintained after submitting.

Is there a requirement for the feature to be active for a period of time before making changes to the settings will take effect?

Or was there just an issue with the form?1

**Comment by createyourpersonalaccount** [comment]
I'm stuck with `Select an option below to continue.` and I can't change any options under **Copilot policies**. For example, **Allow GitHub to use my data for product improvements** is enabled and I cannot disable it.

**Reply by AkshayRaut1321** [reply]
I am not facing problem as the OP but then I read your comment here. I am facing problem same as yours. I started a discussion here because behavior is different. https://github.com/orgs/community/discussions/154716

**Reply by AkshayRaut1321** [reply]
Found the solution, I have updated with my answer - https://github.com/orgs/community/discussions/154716

**Comment by shinyano** [comment]
experiencing same issue here. can't really `select an option`

**Reply by shinyano** [reply]
okay I checked `Suggestions matching public code (duplication detection filter)` and everything seems fine now.

**Comment by AkshayRaut1321** [comment]
Found the solution, I have updated this with my answer - https://github.com/orgs/community/discussions/154716
